In [1]:
# Importing libraries
import pandas as pd

In [2]:
# Reading data
df = pd.read_csv("NAC_1992_2018.csv", sep=";", low_memory=False)

In [3]:
# Creating new data set, where all observations are as recent as year 2000.
millenials = df.loc[df["ANO_NAC"] >= 2000]
millenials.shape

(4552406, 35)

From the shape I can assume that 4552406 - 1 observations are children born -> 4552405. The minus 1 comes from the first row which is the names of the columns.

In [4]:
millenials.head()

ID_RECIEN_NACIDO  SEXO  DIA_NAC  MES_NAC  \
279   3C4318416A599557F6FFA52B8ECC26D35897982E     1     20.0      2.0   
367   55A8B0A2C8315E928226582139B14EB81EB1E4EC     1      3.0      5.0   
1792  FE82B6EC36CDF3B65A7B8FE0ED429DF44A8F137A     1      3.0      2.0   
3221  E3E5E59DCC2CD3D726B51DE8369C2B10466C16A8     2     16.0      2.0   
3364  E2398414C0519B2295369113C933029E0F2BC3FF     1     25.0      2.0   

      ANO_NAC FECHA_NACIMIENTO  TIPO_PARTO  TIPO_ATEN  PARTO_LOCAL  SEMANAS  \
279      2008       20-02-2008           1          1            1     38.0   
367      2008       03-05-2008           1          2            1     38.0   
1792     2008       03-02-2008           1          2            1     39.0   
3221     2008       16-02-2008           1          1            1     37.0   
3364     2008       25-02-2008           1          2            1     40.0   

      ...  CATEG_MADRE  NACIONALIDAD_MADRE  PAIS_ORIGEN_MADRE  \
279   ...          0.0                 NaN                NaN   
367   ...          0.0                 NaN                NaN   
1792  ...          0.0                 NaN                NaN   
3221  ...          0.0                 NaN                NaN   
3364  ...          0.0                 NaN                NaN   

      COMUNA_RESIDENCIA  GLOSA_COMUNA_RESIDENCIA  REGION_RESIDENCIA  \
279                3102                  Caldera                  3   
367                8305                  Mulch�n                  8   
1792               9114               Pitrufqu�n                  9   
3221               8301              Los �ngeles                  8   
3364              10301                   Osorno                 10   

     GLOSA_REGION_RESIDENCIA  SERV_RES GLOSA_SERV_RES  URBANO_RURAL  
279               De Atacama         4        Atacama           1.0  
367               Del B�ob�o        20         B�ob�o           1.0  
1792         De La Araucan�a        21  Araucan�a Sur           1.0  
3221              Del B�ob�o        20         B�ob�o           1.0  
3364            De Los Lagos        23         Osorno           1.0  

[5 rows x 35 columns]

### Important
I come a across some ID_RECIEN_NACIDO that were counted twice. **What do they imply?** Double-counting (error), twin (on purpose) or something different?

In [5]:
# Insight: (Double counting or twins???). If it is double counting then the total number of children is 4491039 (the length of ID_Recien_Nacido)
millenials['ID_RECIEN_NACIDO'].value_counts()

EF7FB0988A5DE4382A2ABE485EA3786D09B671A2    2
4EC68B198A63B1F3B66ECD539C119AA01FF0E533    2
E4D78D1EAD6358538A5E35A15482B908840E3BAE    2
425A1D21F863849BB37DC51840E80560DAADF57F    2
AA69956F0877E91FCB50EFA0DDD18A0BF385898F    2
                                           ..
AE72458F8578D36A95D5FF5952F2385E9372472F    1
BB6EB196B2D86881199504AB96B2923B4E876579    1
2E139AE8071F3600B17BBD9EC9E2AABA730BFAA8    1
70D1B4B13ECE3CB11A5651471D355F71E263F8F8    1
183818A6BA1FE596E4DCE85D97B43B6D88A1F95E    1
Name: ID_RECIEN_NACIDO, Length: 4491039, dtype: int64

In [6]:
# Sorting observations by ascending year, 
millenials.sort_values(by = ['ANO_NAC','COMUNA_RESIDENCIA', 'REGION_RESIDENCIA'], ascending=True)

ID_RECIEN_NACIDO  SEXO  DIA_NAC  MES_NAC  \
779607   8C84C30741432E7C7B96C000E7A51B3B2699258F     2      1.0      8.0   
879623   11790680CD5305320A95B30373444CF752CEB1CC     2      1.0      8.0   
1054424  FDE6EBB763A6E7ADA2305313CE533659EFE4A8F0     2      4.0      8.0   
1294854  5E6A17EA4A3D156AAC1334AFB4058B11F341D9E6     1      2.0      8.0   
1294855  9F73A61CA26BFC1C1E23AE8BFC550B0EB6682CF9     2     11.0      3.0   
...                                           ...   ...      ...      ...   
6326345  B2F775EEDB2509DA8EA1CA1A4923A03E6026036A     2      7.0      7.0   
6513609  F1516610E7ACBE2C1F7C647AF48FAC1788D5AF09     1      1.0     11.0   
6518839  78257B2C4DBF9D9BD2BA7EC2395CB36BD9424010     2     23.0      1.0   
6563551  A8670F30F974DAE8BC1D949A0E5A1681DD502AF6     1      1.0     10.0   
6578320  C0B785A96AD5945F7E32C847950060296A612716     2      1.0     11.0   

         ANO_NAC FECHA_NACIMIENTO  TIPO_PARTO  TIPO_ATEN  PARTO_LOCAL  \
779607      2000       01-08-2000           1          2            1   
879623      2000       01-08-2000           1          1            1   
1054424     2000       04-08-2000           2          1            3   
1294854     2000       02-08-2000           1          2            1   
1294855     2000       11-03-2000           1          2            1   
...          ...              ...         ...        ...          ...   
6326345     2018       07-07-2018           1          2            1   
6513609     2018       01-11-2018           1          2            1   
6518839     2018       23-01-2018           1          1            1   
6563551     2018       01-10-2018           1          1            1   
6578320     2018       01-11-2018           1          2            1   

         SEMANAS  ...  CATEG_MADRE  NACIONALIDAD_MADRE  PAIS_ORIGEN_MADRE  \
779607      37.0  ...          0.0                 NaN                NaN   
879623      37.0  ...          2.0                 NaN                NaN   
1054424     34.0  ...          2.0                 NaN                NaN   
1294854     40.0  ...          0.0                 NaN                NaN   
1294855     36.0  ...          0.0                 NaN                NaN   
...          ...  ...          ...                 ...                ...   
6326345     36.0  ...          0.0                   C                NaN   
6513609     40.0  ...          0.0                   C                NaN   
6518839     38.0  ...          0.0                   C                NaN   
6563551     39.0  ...          2.0                   E              HAIT�   
6578320     41.0  ...          0.0                   C                NaN   

         COMUNA_RESIDENCIA  GLOSA_COMUNA_RESIDENCIA  REGION_RESIDENCIA  \
779607                1101                  Iquique                  1   
879623                1101                  Iquique                  1   
1054424               1101                  Iquique                  1   
1294854               1101                  Iquique                  1   
1294855               1101                  Iquique                  1   
...                    ...                      ...                ...   
6326345              16305              San Nicol�s                 16   
6513609              16305              San Nicol�s                 16   
6518839              16305              San Nicol�s                 16   
6563551              16305              San Nicol�s                 16   
6578320              16305              San Nicol�s                 16   

        GLOSA_REGION_RESIDENCIA  SERV_RES GLOSA_SERV_RES  URBANO_RURAL  
779607              De Tarapac�         2        Iquique           1.0  
879623              De Tarapac�         2        Iquique           1.0  
1054424             De Tarapac�         2        Iquique           1.0  
1294854             De Tarapac�         2        Iquique           1.0  
1294855             De Tarapac�         2        Iqu

In [7]:
millenials = millenials.reset_index()

In [8]:
millenials.head()

index                          ID_RECIEN_NACIDO  SEXO  DIA_NAC  MES_NAC  \
0    279  3C4318416A599557F6FFA52B8ECC26D35897982E     1     20.0      2.0   
1    367  55A8B0A2C8315E928226582139B14EB81EB1E4EC     1      3.0      5.0   
2   1792  FE82B6EC36CDF3B65A7B8FE0ED429DF44A8F137A     1      3.0      2.0   
3   3221  E3E5E59DCC2CD3D726B51DE8369C2B10466C16A8     2     16.0      2.0   
4   3364  E2398414C0519B2295369113C933029E0F2BC3FF     1     25.0      2.0   

   ANO_NAC FECHA_NACIMIENTO  TIPO_PARTO  TIPO_ATEN  PARTO_LOCAL  ...  \
0     2008       20-02-2008           1          1            1  ...   
1     2008       03-05-2008           1          2            1  ...   
2     2008       03-02-2008           1          2            1  ...   
3     2008       16-02-2008           1          1            1  ...   
4     2008       25-02-2008           1          2            1  ...   

   CATEG_MADRE  NACIONALIDAD_MADRE  PAIS_ORIGEN_MADRE  COMUNA_RESIDENCIA  \
0          0.0                 NaN                NaN               3102   
1          0.0                 NaN                NaN               8305   
2          0.0                 NaN                NaN               9114   
3          0.0                 NaN                NaN               8301   
4          0.0                 NaN                NaN              10301   

   GLOSA_COMUNA_RESIDENCIA  REGION_RESIDENCIA  GLOSA_REGION_RESIDENCIA  \
0                  Caldera                  3               De Atacama   
1                  Mulch�n                  8               Del B�ob�o   
2               Pitrufqu�n                  9          De La Araucan�a   
3              Los �ngeles                  8               Del B�ob�o   
4                   Osorno                 10             De Los Lagos   

  SERV_RES  GLOSA_SERV_RES URBANO_RURAL  
0        4         Atacama          1.0  
1       20          B�ob�o          1.0  
2       21   Araucan�a Sur          1.0  
3       20          B�ob�o          1.0  
4       23          Osorno          1.0  

[5 rows x 36 columns]

In [10]:
# [TODO]: Based on Britta's response deal with INSIGHT SOLVE
millenials['INSIGHT SOLVE'] = millenials.groupby(by='ID_RECIEN_NACIDO')['ID_RECIEN_NACIDO'].transform('count')
millenials

index                          ID_RECIEN_NACIDO  SEXO  DIA_NAC  \
0            279  3C4318416A599557F6FFA52B8ECC26D35897982E     1     20.0   
1            367  55A8B0A2C8315E928226582139B14EB81EB1E4EC     1      3.0   
2           1792  FE82B6EC36CDF3B65A7B8FE0ED429DF44A8F137A     1      3.0   
3           3221  E3E5E59DCC2CD3D726B51DE8369C2B10466C16A8     2     16.0   
4           3364  E2398414C0519B2295369113C933029E0F2BC3FF     1     25.0   
...          ...                                       ...   ...      ...   
4552401  6679527  632A8859668611B334BB461A52FB23F42D0396BF     2     11.0   
4552402  6679528  763D2F12FB28DB06015FAC8E9ED3CDC96DA83635     1     10.0   
4552403  6679529  8B0AF2D5E30331343E6C9B363425280FE4592ED1     1     10.0   
4552404  6679530  3D52DA8B577A0C94A513AD24791AD6EAB48F46CC     1     23.0   
4552405  6679531  FB50E15FA0588296E56ACB84C39CBACD643C77AF     2     14.0   

         MES_NAC  ANO_NAC FECHA_NACIMIENTO  TIPO_PARTO  TIPO_ATEN  \
0            2.0     2008       20-02-2008           1          1   
1            5.0     2008       03-05-2008           1          2   
2            2.0     2008       03-02-2008           1          2   
3            2.0     2008       16-02-2008           1          1   
4            2.0     2008       25-02-2008           1          2   
...          ...      ...              ...         ...        ...   
4552401     10.0     2018       11-10-2018           1          1   
4552402      9.0     2018       10-09-2018           1          2   
4552403     10.0     2018       10-10-2018           1          1   
4552404      3.0     2018       23-03-2018           1          2   
4552405     12.0     2018       14-12-2018           1          2   

         PARTO_LOCAL  ...                         PAIS_ORIGEN_MADRE  \
0                  1  ...                                       NaN   
1                  1  ...                                       NaN   
2                  1  ...                                       NaN   
3                  1  ...                                       NaN   
4                  1  ...                                       NaN   
...              ...  ...                                       ...   
4552401            1  ...                                 VENEZUELA   
4552402            1  ...                                     HAIT�   
4552403            1  ...                                     HAIT�   
4552404            1  ...  PER�                                       
4552405            1  ...                                      PER�   

         COMUNA_RESIDENCIA  GLOSA_COMUNA_RESIDENCIA  REGION_RESIDENCIA  \
0                     3102                  Caldera                  3   
1                     8305                  Mulch�n                  8   
2                     9114               Pitrufqu�n                  9   
3                     8301              Los �ngeles                  8   
4                    10301                   Osorno                 10   
...                    ...                      ...                ...   
4552401               8101               Concepci�n                  8   
4552402              13116                Lo Espejo                 13   
4552403              13402                     Buin                 13   
4552404               6101                 Rancagua                  6   
4552405              13119                    Maip�                 13   

             GLOSA_REGION_RESIDENCIA  SERV_RES           GLOSA_SERV_RES  \
0                         De Atacama         4                  Atacama   
1                         Del B�ob�o        20                   B�ob�o   
2                    De La Araucan�a        21            Araucan�a Sur   
3                         Del B�ob�o        20                   B�ob�o   
4                       De Los Lagos        23                   Osorno   
...                              ...       ...                   

In [11]:
millenials.drop('total',axis=1)

index                          ID_RECIEN_NACIDO  SEXO  DIA_NAC  \
0            279  3C4318416A599557F6FFA52B8ECC26D35897982E     1     20.0   
1            367  55A8B0A2C8315E928226582139B14EB81EB1E4EC     1      3.0   
2           1792  FE82B6EC36CDF3B65A7B8FE0ED429DF44A8F137A     1      3.0   
3           3221  E3E5E59DCC2CD3D726B51DE8369C2B10466C16A8     2     16.0   
4           3364  E2398414C0519B2295369113C933029E0F2BC3FF     1     25.0   
...          ...                                       ...   ...      ...   
4552401  6679527  632A8859668611B334BB461A52FB23F42D0396BF     2     11.0   
4552402  6679528  763D2F12FB28DB06015FAC8E9ED3CDC96DA83635     1     10.0   
4552403  6679529  8B0AF2D5E30331343E6C9B363425280FE4592ED1     1     10.0   
4552404  6679530  3D52DA8B577A0C94A513AD24791AD6EAB48F46CC     1     23.0   
4552405  6679531  FB50E15FA0588296E56ACB84C39CBACD643C77AF     2     14.0   

         MES_NAC  ANO_NAC FECHA_NACIMIENTO  TIPO_PARTO  TIPO_ATEN  \
0            2.0     2008       20-02-2008           1          1   
1            5.0     2008       03-05-2008           1          2   
2            2.0     2008       03-02-2008           1          2   
3            2.0     2008       16-02-2008           1          1   
4            2.0     2008       25-02-2008           1          2   
...          ...      ...              ...         ...        ...   
4552401     10.0     2018       11-10-2018           1          1   
4552402      9.0     2018       10-09-2018           1          2   
4552403     10.0     2018       10-10-2018           1          1   
4552404      3.0     2018       23-03-2018           1          2   
4552405     12.0     2018       14-12-2018           1          2   

         PARTO_LOCAL  ...  NACIONALIDAD_MADRE  \
0                  1  ...                 NaN   
1                  1  ...                 NaN   
2                  1  ...                 NaN   
3                  1  ...                 NaN   
4                  1  ...                 NaN   
...              ...  ...                 ...   
4552401            1  ...                   E   
4552402            1  ...                   E   
4552403            1  ...                   E   
4552404            1  ...                   E   
4552405            1  ...                   E   

                                PAIS_ORIGEN_MADRE  COMUNA_RESIDENCIA  \
0                                             NaN               3102   
1                                             NaN               8305   
2                                             NaN               9114   
3                                             NaN               8301   
4                                             NaN              10301   
...                                           ...                ...   
4552401                                 VENEZUELA               8101   
4552402                                     HAIT�              13116   
4552403                                     HAIT�              13402   
4552404  PER�                                                   6101   
4552405                                      PER�              13119   

         GLOSA_COMUNA_RESIDENCIA  REGION_RESIDENCIA  \
0                        Caldera                  3   
1                        Mulch�n                  8   
2                     Pitrufqu�n                  9   
3                    Los �ngeles                  8   
4                         Osorno                 10   
...                          ...                ...   
4552401               Concepci�n                  8   
4552402                Lo Espejo                 13   
4552403                     Buin                 13   
4552404                 Rancagua                  6   
4552405                    Maip�                 13   

             GLOSA_REGION_RESIDENCIA  SERV_RES           GLOSA_SERV_RES  \
0                         De Atacama         4                  Ata

In [22]:
subset = millenials.groupby(['ANO_NAC', 'GLOSA_REGION_RESIDENCIA', 'GLOSA_COMUNA_RESIDENCIA']).agg('count')

In [24]:
# Cleaning
target = list(subset)
new_target = target.pop(1)
target

['index',
 'SEXO',
 'DIA_NAC',
 'MES_NAC',
 'FECHA_NACIMIENTO',
 'TIPO_PARTO',
 'TIPO_ATEN',
 'PARTO_LOCAL',
 'SEMANAS',
 'PESO',
 'TALLA',
 'EDAD_PADRE',
 'CURSO_PADRE',
 'NIVEL_PADRE',
 'ACTIV_PADRE',
 'OCUPA_PADRE',
 'CATEG_PADRE',
 'ID_MADRE',
 'EDAD_MADRE',
 'EST_CIV_MADRE',
 'CURSO_MADRE',
 'NIVEL_MADRE',
 'ACTIV_MADRE',
 'OCUPA_MADRE',
 'CATEG_MADRE',
 'NACIONALIDAD_MADRE',
 'PAIS_ORIGEN_MADRE',
 'COMUNA_RESIDENCIA',
 'REGION_RESIDENCIA',
 'SERV_RES',
 'GLOSA_SERV_RES',
 'URBANO_RURAL',
 'total',
 'INSIGHT SOLVE']

In [25]:
final = subset.drop(target, axis=1).rename(columns={"ID_RECIEN_NACIDO": "Total births"})
final.head()

Total births
ANO_NAC GLOSA_REGION_RESIDENCIA                GLOSA_COMUNA_RESIDENCIA              
2000    De Ais�n del Gral. C. Ib��ez del Campo Ais�n                             316
                                               Chile Chico                        66
                                               Cisnes                             95
                                               Cochrane                           54
                                               Coihaique                         899

In [26]:
final.shape

(6507, 1)

In [27]:
final['Total births'].sum()

4491115

In [29]:
 4491115 - 4491039

76

**Assumming they have double counted:** 
Result most likely to be true, with: 4491115 - 4491039 (Total number of different IDs) = 76 observations too much. These are scattered across different comunas). 

In [32]:
final.to_csv('childborn.csv') 

In [33]:
final.to_excel('childborn.xls') 